### ISSUE TITLE
Manifest migration from Helm values fails in 1.5.0

### ISSUE THREAD

**Issue by seeruk** [istio/istio#22089]
**Bug description**

Running `istioctl manifest migrate` on some Istio Helm values using `istioctl` 1.5.0 outputs the following for me:

```
$ istioctl manifest migrate ./values-staging.yaml
proto: tag has too few fields: "-"
Error: the input values.yaml fail validation: json: cannot unmarshal object into Go value of type []json.RawMessage
check against https://github.com/istio/istio/blob/releaese-1.5/operator/pkg/apis/istio/v1alpha1/values_types.proto for schema
or run the command with --force flag to ignore the error
```

If I used `istioctl` from an older version (the previous version I had was 1.4.3), it works:

```
$ /opt/istio-1.4.3/bin/istioctl manifest migrate ./values-staging.yaml
proto: tag has too few fields: "-"
apiVersion: install.istio.io/v1alpha2
kind: IstioControlPlane
metadata:
  creationTimestamp: null
spec:
  autoInjection:
    components:
      injector:
...
```

If I take the `IstioControlPlane` output from the 1.4.3 `istioctl` and feed that to the 1.5.0 `istioctl` that works (but the produced file actually doesn't work; #22088, #21938 and #20033)

**Expected behavior**

`istioctl` should still be able to translate Helm values (it still says it should in the help text at least!)

**Steps to reproduce the bug**

The `values.yaml` used (with a swapped out IP range in the ingress gateway) is this: https://gist.github.com/seeruk/c59c42fe9df7608c137883e8c0e82b39

**Version (include the output of `istioctl version --remote` and `kubectl version` and `helm version` if you used Helm)**

```
$ istioctl version --remote
client version: 1.5.0
citadel version: 1.5.0
citadel version: 1.5.0
galley version: 1.5.0
galley version: 1.5.0
ilbgateway version: 1.5.0
ilbgateway version: 1.5.0
ingressfirewalledgateway version:
ingressfirewalledgateway version:
ingressgateway version: 1.5.0
ingressgateway version: 1.5.0
pilot version: 1.5.0
pilot version: 1.5.0
policy version: 1.5.0
policy version: 1.5.0
sidecar-injector version: 1.5.0
sidecar-injector version: 1.5.0
telemetry version: 1.5.0
telemetry version: 1.5.0
data plane version: 1.5.0 (87 proxies)
```

```
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.3", GitCommit:"06ad960bfd03b39c8310aaf92d1e7c12ce618213", GitTreeState:"clean", BuildDate:"2020-02-13T18:08:14Z", GoVersion:"go1.13.8", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"14+", GitVersion:"v1.14.9-eks-502bfb", GitCommit:"502bfb383169b124d87848f89e17a04b9fc1f6f0", GitTreeState:"clean", BuildDate:"2020-02-07T01:31:02Z", GoVersion:"go1.12.12", Compiler:"gc", Platform:"linux/amd64"}
```

```
$ helm version
Client: &version.Version{SemVer:"v2.16.1", GitCommit:"bbdfe5e7803a12bbdf97e94cd847859890cf4050", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.16.1", GitCommit:"bbdfe5e7803a12bbdf97e94cd847859890cf4050", GitTreeState:"clean"}
```

**How was Istio installed?**

Via Helm

**Environment where bug was observed (cloud vendor, OS, etc)**

AWS EKS

**Comment by richardwxn** [comment]
@seeruk the reason why you saw this error "Error: the input values.yaml fail validation" is because your input values.yaml file has outdated schema, which means that the output IstioOperator spec would contain fields that are not supported, that's why it fails and generate the message suggest you to check against the schema of corresponding version. You can always use --force flag to proceed but the risk is that the outdated settings may not work as expected.

If you don't mind, can you paste your input values.yaml so I can take a look?

**Comment by seeruk** [comment]
Aah, that's interesting. I'm a bit confused though because the [values.yaml](https://gist.github.com/seeruk/c59c42fe9df7608c137883e8c0e82b39) I used (linked) works when used with the Istio 1.5.0 Helm Chart just fine.

**Comment by seeruk** [comment]
Hey @richardwxn, appreciate you looking into this one - I found the particular part of the values that was causing the issue and removed it. Unfortunately, it then presents me with a different issue related to the multiple gateways being unexpected:

```
$ istioctl manifest migrate ./values-staging.yaml
proto: tag has too few fields: "-"
Error: the input values.yaml fail validation: unknown field "istio-ingressfirewalledgateway" in v1alpha1.GatewaysConfig
check against https://github.com/istio/istio/blob/releaese-1.5/operator/pkg/apis/istio/v1alpha1/values_types.proto for schema
or run the command with --force flag to ignore the error
```

From looking at the schema I can see it expects just the one `istio-ingressgateway` entry. I spent this afternoon figuring out how to manually put together a working Ingress Gateway by manually creating the required Kubernetes and Istio resources, so that's an option I can probably use for now until multiple gateways are supported by `istioctl` (hopefully the next version includes the change to support generating the resources without hard-coded names - I see this one has been merged, and with support for loadBalancerSourceRanges and other service configuration).
